backend/src/nodes/utils/ncnn_model.py

```diff
@@ -0,0 +1,524 @@
+                param_schema[op_type][ks]["paramPhase"],
+                v,
+                param_schema[op_type][ks]["defaultValue"],
+            )
+            param_dict[k] = param
+
+        return op_type, NcnnLayer(
+            op_type,
+            name,
+            num_inputs,
+            num_outputs,
+            inputs,
+            outputs,
+            NcnnParamCollection(op_type, param_dict),
+        )
+
+    def load_layer_weights(
+        self, binf: BufferedReader, op_type: str, layer: NcnnLayer
+    ) -> Dict[str, NcnnWeight]:
+        weight_dict = {}
+        if op_type == "Convolution":
+            quantize_tag = binf.read(4)
+            dtype = np.float16 if quantize_tag == DTYPE_FP16 else np.float32
+            weight_data_length = layer.params[6].value
+            assert isinstance(weight_data_length, int), "Data size must be int"
+            weight_data_size = (
+                weight_data_length * 2
+                if quantize_tag == DTYPE_FP16
+                else weight_data_length * 4
+            )
+
+            has_bias = layer.params[5].value if 5 in layer.params else 0
+
+            num_filters = layer.params[0].value
+            kernel_w = layer.params[1].value
+            kernel_h = layer.params[11].value if 11 in layer.params else kernel_w
+            num_input = weight_data_length // num_filters // kernel_w // kernel_h  # type: ignore
+            shape = (num_filters, num_input, kernel_w, kernel_h)
+
+            weight_data = np.frombuffer(binf.read(weight_data_size), dtype)
+            weight_data = weight_data.reshape(shape)  # type: ignore
+            weight_dict["weight"] = NcnnWeight(weight_data, quantize_tag)
+
+            if has_bias:
+                bias_data_size = num_filters * 4
+                bias_data = np.frombuffer(binf.read(bias_data_size), np.float32)  # type: ignore
+                weight_dict["bias"] = NcnnWeight(bias_data)
+
+        return weight_dict
+
+    def load_model(self, param_path: str, bin_path: Union[str, None] = None) -> None:
+        if bin_path is None:
+            bin_path = param_path.replace(".param", ".bin")
+
+        assert os.path.exists(param_path), f"{param_path} does not exist"
+        assert os.path.exists(bin_path), f"{bin_path} does not exist"
+
+        with open(param_path, "r") as paramf:
+            with open(bin_path, "rb") as binf:
+                paramf.readline()
+                counts = paramf.readline().strip().split(" ")
+                self.node_count = int(counts[0])
+                self.blob_count = int(counts[1])
+
+                for line in paramf:
+                    op_type, layer = self.parse_param_layer(line)
+                    layer.weight_data = self.load_layer_weights(binf, op_type, layer)
+                    self.add_layer(layer)
+
+    def write_param(self, filename: str) -> None:
+        with open(filename, "w", encoding="utf-8") as f:
+            f.write(f"{self.MAGIC}\n")
```

Comment:
This feels like it should be a static constant and not a mutable instance field.

Reply:
Should be able to do something like 
```py
    @property
    def magic(cls):
        return "7767517"
```
Which would make it immutable

Reply:
But that's still on instance thing.

Reply:
(deleted the original message since it put it on its own in my review)
Should be able to do something like 
```py
    @property
    def magic(cls):
        return "7767517"
```
Which would make it immutable